sklearn/utils/estimator_checks.py

```diff
@@ -1207,7 +1204,7 @@ def check_methods_sample_order_invariance(
 
 
 @ignore_warnings
-def check_fit2d_1sample(name, estimator_orig, strict_mode=True):
+def check_fit2d_1sample(name, estimator_orig, api_only=False):
```

Comment:
so your point is making sure that giving certain input type doesn't raise is not test we want to have? Very often people assume the data is of the format they have expected in their scripts, and forget to check for other types (like calling `check_X_y`, etc), and these tests make sure they do these validations.

Reply:
some of the dtype checking is covered with other tests though, hmm....

Reply:
This check isn't about dtypes or input types. It's about passing one single sample to fit().

Here, the only way to make this check less strict is to not validate the error message. But then this test would become completely useless: it would make sure that something either passes or that it doesn't.

Re dtypes: I've put `check_estimators_dtypes` as a pure API check as you suggested before, but I don't understand how that relates here.